src/include/sof/spinlock.h

```diff
@@ -81,44 +81,46 @@ extern uint32_t lock_dbg_atomic;
 extern uint32_t lock_dbg_user[DBG_LOCK_USERS];
 
 /* all SMP spinlocks need init, nothing todo on UP */
-#define spinlock_init(lock) \
-	do { \
-		arch_spinlock_init(lock); \
-		(*lock)->user = __LINE__; \
-	} while (0)
+static inline void spinlock_init(spinlock_t *lock)
+{
+	arch_spinlock_init(lock);
+	(*lock)->user = __LINE__;
+}
 
 /* panic on deadlock */
-#define spin_try_lock_dbg(lock) \
-	do { \
-		int __tries; \
-		for (__tries = DBG_LOCK_TRIES;  __tries > 0; __tries--) { \
-			if (arch_try_lock(lock)) \
-				break;	/* lock acquired */ \
-		} \
-		if (__tries == 0) { \
-			trace_lock_error("DED"); \
-			trace_lock_value(__LINE__); \
-			trace_lock_value((lock)->user); \
-			panic(SOF_IPC_PANIC_DEADLOCK); /* lock not acquired */ \
-		} \
-	} while (0)
+static inline void spin_try_lock_dbg(spinlock_t *lock)
+{
+	int __tries;
+
+	for (__tries = DBG_LOCK_TRIES; __tries > 0; __tries--) {
+		if (arch_try_lock(lock))
+			break;	/* lock acquired */
+	}
+	if (__tries == 0) {
+		trace_lock_error("DED");
+		trace_lock_value(__LINE__);
```

Comment:
Oof, line numbers will be messed up when using inline functions. More thought needs to go into this. For now you can just remove the line number because it'll be incorrect anyway.